docs/sources/trackers/javascript-trackers/web-tracker/cross-domain-tracking/index.md

```diff
@@ -4,23 +4,351 @@ date: "2022-08-30"
+});
+```
 
   </TabItem>
 </Tabs>
+
+
+### Decorate all links
+
+To decorate every link, regardless of its destination:
+
+```javascript
+{
+  crossDomainLinker: function (linkElement) {
+    return true;
+  }
+}
+```
+
+This will decorate "links" that are actually just JavaScript actions (with an `href` of `"javascript:void(0)"`), or links to email addresses or telephone numbers (`mailto:` or `tel:` schemes). To exclude these links, check them explicitly:
+
+<Tabs groupId="platform" queryString>
+  <TabItem value="js" label="JavaScript (tag)" default>
+
+```javascript
+snowplow('crossDomainLinker', function(linkElement) {
+  return linkElement.href.indexOf('javascript:') < 0;
+});
+```
+
+  </TabItem>
+  <TabItem value="browser" label="Browser (npm)">
+
+```javascript
+crossDomainLinker(function(linkElement) {
+  return linkElement.href.indexOf('javascript:') < 0;
+});
+```
+
+  </TabItem>
+</Tabs>
+
+Alternatively, only count links that parse as web URLs by checking the link's [`hostname`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/hostname). This should automatically exclude links that don't lead simply to other web pages.
+
+<Tabs groupId="platform" queryString>
+  <TabItem value="js" label="JavaScript (tag)" default>
+
+```javascript
+snowplow('crossDomainLinker', function(linkElement) {
+  return linkElement.hostname !== "";
+});
+```
+
+  </TabItem>
+  <TabItem value="browser" label="Browser (npm)">
+
+```javascript
+crossDomainLinker(function(linkElement) {
+  return linkElement.hostname !== "";
+});
+```
+
+  </TabItem>
+</Tabs>
+
+
+### Opt-in and opt-out
+
+Some of the examples will also decorate any links to the same site the user is currently on.
+
+We recommend explicitly allowing the list of domains you want to decorate. Only decorate links that are on a different domain to the current page.
+
+```javascript
+{
+  crossDomainLinker: function (linkElement) {
+    return linkElement.hostname !== location.hostname;
+  }
+}
+```
+
+If you have a shared configuration deployed across multiple sites, you may want to combine the approaches:
+
+```javascript
+{
+  crossDomainLinker: function (linkElement) {
+    var networkSites = [
+      "mysite.com",
+      "sistersite.com"
+    ];
+    return linkElement.hostname !== location.hostname && networkSites.indexOf(linkElement.hostname) > -1;
+  }
+}
+```
+
+Link decoration will only occur when links are to different sites within the list.
+
+### Update event listeners
+
+When the tracker loads it doesn't immediately decorate links. Instead, it adds event listeners to links which decorate them when a user clicks on them or navigates to them using the keyboard. This ensures that the timestamp added to the querystring is fresh.
+
+If further links get added to the page after the tracker has loaded, you can use the tracker's `crossDomainLinker` method to add listeners again. Listeners won't be added to links which already have them.
```

Comment:
... so it's good practice to make sure you use the same linker function every call, it won't be updated.